src/libServer/LookupServer.cpp

```diff
@@ -1357,17 +1371,17 @@ Json::Value LookupServer::GetEthTransactionReceipt(const std::string& txnhash) {
     std::string toAddr = ethResult["to"].asString();
     std::string cumGas = zilResult["cumulative_gas"].asString();
 
-    if (blockHash == "") {
-      LOG_GENERAL(WARNING, "Tx receipt requested but not found in any blocks.");
-      return "";
-    }
-
     if (blockHash.size() > 2 && blockHash[0] != '0' && blockHash[1] != 'x') {
       blockHash = std::string("0x") + blockHash;
     }
 
-    auto res = populateReceiptHelper(hashId, success, sender, toAddr, cumGas,
-                                     blockHash, blockNumber);
+    // TODO: check if EVM-DS expects constructor calldata to be passed in 'data'
+    // field.
```

Comment:
I can asnwer the question in this TODO:  It does expect calldata to be passed in the data field.

But hey, you cannot distinguish contract deployment or not by just looking at the data field.
You can easily have a contract call with an empty calldata. In fact, this is quite common.

The right check is destination address being zero. See `Transaction::GetTransactionType()`